### DISCUSSION TITLE
How can i deserialize PreparsedDocumentEntry

### DISCUSSION THREAD

**Discussion by SaranSundar** [Netflix/dgs-framework#1601] [Q&A]
https://netflix.github.io/dgs/advanced/operation-caching/

```
@Override
    public PreparsedDocumentEntry getDocument(ExecutionInput executionInput,
                                              Function<ExecutionInput, PreparsedDocumentEntry> parseAndValidateFunction) {
        return cache.get(executionInput.getQuery(), operationString -> parseAndValidateFunction.apply(executionInput));
    }
```

I'm trying to save the query string and value to redis and read it back, 

but while PreparsedDocumentEntry implements Serialiazble to be able to write it to redis, it has no default constructor so Jackson fails when deserializing it with object mapper.

Can we create default constructors for this so it can be deserializable?

**Comment by SaranSundar** [comment]
@srinivasankavitha any thoughts on this?

**Comment by srinivasankavitha** [comment]
PreparsedDocumentEntry is from graphql-java library that the DGS
framework is based on, so this would need to be a request for the library.
The framework simply exposes the functionality for you to use. Hope that
helps.

**Reply by SaranSundar** [reply]
Thanks that makes sense.
